### PR TITLE
Update on timer interval

### LINKED ISSUE #14: Homekit only updates sensor values when you open the app
Right now the plugin only queries for updated API data when Homekit asks for an update (e.g. when you open the Home app, it tries to refresh the status of all your devices). This means that automations that you make that trigger when a sensor moves above or below a threshold will not fire until you open the Home app to check the status of your home. 

Solution: The plugin should periodically request updated API data on its own and push that data to Homekit using the Characteristic.updateValue() function. Home hubs can then properly fire automations based on the sensor data.

I've made a [fork](https://github.com/ninjas28/homebridge-airthings/tree/update_on_timer) that does this, it seems to work except for some weird warning I get:
```This plugin generated a warning from the characteristic 'Status Active': Characteristic not in required or optional characteristic section for service Service. Adding anyway..```
I'll try and figure out why its saying that before I make a PR.

### PR DESCRIPTION
**Pull request by ninjas28**
This fixes #14.
See what you think of this, this is a pretty major rewrite of the entire plugin.
I've been running it for about a week now and its been rock solid.

### CODE REVIEW

**Comment by michaelahern** [comment]
Looks great from a quick read through! Give me a bit of time to run it on my end as well, and will post some comments in the meantime..

File: README.md [hunk 1]
```diff
@@ -41,20 +41,22 @@ Example accessory config in the Homebridge config.json:
...
-**clientSecret**	| (required) API Client Secret generated in the [Airthings Dashboard](https://dashboard.airthings.com)
-**serialNumber**	| (required) Serial number of the device
+Field           	  | Description
+--------------------|------------
+**accessory**   	  | (required) Must be "Airthings"
+**name**					  | (required) Name for the device in HomeKit
+**clientId**			  | (required) API Client ID generated in the [Airthings Dashboard](https://dashboard.airthings.com)
+**clientSecret**	  | (required) API Client Secret generated in the [Airthings Dashboard](https://dashboard.airthings.com)
+**serialNumber**	  | (required) Serial number of the device
+**refreshInterval**	| (required) Refresh interval for grabbing new data in seconds
```
Comment (michaelahern) [inline README.md:59]: Could this be flagged as optional, defaulting to 5 minutes?

File: config.schema.json [hunk 2]
```diff
@@ -25,6 +25,11 @@
         "title": "Serial Number",
         "type": "string",
         "required": true
+      },
+      "refreshInterval": {
+        "title": "Refresh Interval (seconds)",
+        "type": "number",
+        "required": true
```
Comment (michaelahern) [inline config.schema.json:32]: Could this be flagged as optional, with a [placeholder value](https://developers.homebridge.io/#/config-schema#placeholder-values) of 5 minutes?

File: src/plugin.ts [hunk 3]
```diff
@@ -51,6 +59,7 @@ class AirthingsPlugin implements AccessoryPlugin {
 
     this.log.info(`Device Model: ${this.airthingsDevice.model}`);
     this.log.info(`Serial Number: ${config.serialNumber}`);
+    this.log.info(`Refresh Interval: ${config.refreshInterval}`)
```
Comment (michaelahern) [inline src/plugin.ts:62]: Append "s" so the log shows the value is in seconds.

**Review by michaelahern — CHANGES_REQUESTED** [review]

**Comment by ninjas28** [comment]
How's this look?

**Comment by ninjas28** [comment]
Have you had a chance to test this out?

FWIW I've been running this for the last couple months and its worked flawlessly.

**Comment by michaelahern** [comment]
Apologies, took a bit of time, but finally got this merged in. Thanks for the contrib!

**Pull request merged**